packages/vite/src/node/build.ts

```diff
@@ -1031,41 +1028,84 @@ export function onRollupWarning(
         }
       }
 
-      if (
-        warning.plugin === 'rollup-plugin-dynamic-import-variables' &&
-        dynamicImportWarningIgnoreList.some((msg) =>
-          warning.message.includes(msg),
-        )
-      ) {
-        return
-      }
+      if (logLeveling === 'warn') {
+        if (
+          logging.plugin === 'rollup-plugin-dynamic-import-variables' &&
+          dynamicImportWarningIgnoreList.some((msg) =>
+            logging.message.includes(msg),
+          )
+        ) {
+          return
+        }
 
-      if (warningIgnoreList.includes(warning.code!)) {
-        return
+        if (warningIgnoreList.includes(logging.code!)) {
+          return
+        }
       }
 
-      if (warning.code === 'PLUGIN_WARNING') {
-        environment.logger.warn(
-          `${colors.bold(
-            colors.yellow(`[plugin:${warning.plugin}]`),
-          )} ${colors.yellow(warning.message)}`,
-        )
-        return
+      if (
+        logging.code === 'PLUGIN_LOG' ||
+        logging.code === 'PLUGIN_WARNING' ||
+        logging.code === 'PLUGIN_ERROR'
+      ) {
+        switch (logLeveling) {
+          case 'info':
+            environment.logger.info(logging.message)
+            return
+          case 'warn':
+            environment.logger.warn(colors.yellow(logging.message))
+            return
+          case 'error':
+            environment.logger.error(colors.red(logging.message))
+            return
+          case 'debug':
+            debugLogger?.(logging.message)
+            return
+          default:
+            break
+        }
       }
     }
 
-    warn(warnLog)
+    logHandler(logLeveling, logging)
```

Comment:
I wonder if we should always call `environment.logger` instead of calling the default log handler of rollup when `logging.code` is not `PLUGIN_*`.

Reply:
In the team meeting, we decided to always call `environment.logger`.